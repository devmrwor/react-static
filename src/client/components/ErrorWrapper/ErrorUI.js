import React from 'react'
import PropTypes from 'prop-types'

const ErrorUI = ({ error, errorInfo }) => (
  <div
    style={{
      margin: '1rem',
      padding: '1rem',
      background: 'rgba(0,0,0,0.05)',
    }}
  >
    <h2>Oh-no! Somethingâ€™s gone wrong!</h2>
    <pre style={{ whiteSpace: 'normal', color: 'red' }}>
      <code>{error && error.toString()}</code>
    </pre>
    <h3>This error occurred here:</h3>
    <pre style={{ color: 'red', overflow: 'auto' }}>
      <code>{errorInfo.componentStack}</code>
    </pre>
    <p>For more information, please see the console.</p>
  </div>
)

ErrorUI.propTypes = {
  error: PropTypes.object.isRequired,
  errorInfo: PropTypes.object.isRequired,
}

export default ErrorUI
