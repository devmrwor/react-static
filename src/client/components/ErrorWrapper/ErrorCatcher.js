import React from 'react'
import PropTypes from 'prop-types'
import ErrorUI from './ErrorUI'

export default class ErrorCatcher extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  }

  state = {
    error: null,
    errorInfo: null,
  }

  componentDidCatch (error, errorInfo) {
    // Catch errors in any child components and re-renders with an error message
    this.setState({
      error,
      errorInfo,
    })
  }

  render () {
    const { error } = this.state

    // Fallback UI if an error occurs
    if (error) {
      return <ErrorUI {...this.state} />
    }

    return React.Children.only(this.props.children)
  }
}
