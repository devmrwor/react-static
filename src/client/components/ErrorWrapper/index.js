import React from 'react'
import PropTypes from 'prop-types'
import ErrorCatcher from './ErrorCatcher'

const ErrorWrapper = ({ showErrorsInProduction, children }) => {
  if (process.env.REACT_STATIC_ENV === 'development' || showErrorsInProduction) {
    return <ErrorCatcher>{children}</ErrorCatcher>
  }

  return React.Children.only(children)
}

ErrorWrapper.propTypes = {
  showErrorsInProduction: PropTypes.bool,
  children: PropTypes.node.isRequired,
}

ErrorWrapper.defaultProps = {
  showErrorsInProduction: false,
}

export default ErrorWrapper
