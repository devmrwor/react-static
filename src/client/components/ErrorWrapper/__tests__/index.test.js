import React from 'react'
import { shallow } from 'enzyme'
import ErrorWrapper from '../index'

describe('ErrorWrapper', () => {
  let reactStaticEnviroment

  beforeEach(() => {
    reactStaticEnviroment = process.env.REACT_STATIC_ENV
  })

  describe('when process.env.REACT_STATIC_ENV is `development`', () => {
    it('should wrap the child with the ErrorCatcher', () => {
      process.env.REACT_STATIC_ENV = 'development'

      const errorWrapper = shallow(
        <ErrorWrapper>
          <span>hello</span>
        </ErrorWrapper>
      )

      expect(errorWrapper).toMatchSnapshot()
    })
  })

  describe('when process.env.REACT_STATIC_ENV is `production`', () => {
    it('should not wrap the child with the ErrorCatcher', () => {
      process.env.REACT_STATIC_ENV = 'production'

      const errorWrapper = shallow(
        <ErrorWrapper >
          <span>hello</span>
        </ErrorWrapper>
      )

      expect(errorWrapper).toMatchSnapshot()
    })

    describe('when showErrorsInProduction is defined', () => {
      it('should wrap the child with the ErrorCatcher', () => {
        process.env.REACT_STATIC_ENV = 'production'

        const errorWrapper = shallow(
          <ErrorWrapper showErrorsInProduction>
            <span>hello</span>
          </ErrorWrapper>
        )

        expect(errorWrapper).toMatchSnapshot()
      })
    })
  })

  afterEach(() => {
    process.env.REACT_STATIC_ENV = reactStaticEnviroment
  })
})
