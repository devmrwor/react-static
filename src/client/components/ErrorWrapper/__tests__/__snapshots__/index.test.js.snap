// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `ErrorWrapper when process.env.REACT_STATIC_ENV is \`development\` should wrap the child with the ErrorCatcher 1`
] = `
<ErrorCatcher>
  <span>
    hello
  </span>
</ErrorCatcher>
`

exports[
  `ErrorWrapper when process.env.REACT_STATIC_ENV is \`production\` should not wrap the child with the ErrorCatcher 1`
] = `
<span>
  hello
</span>
`

exports[
  `ErrorWrapper when process.env.REACT_STATIC_ENV is \`production\` when showErrorsInProduction is defined should wrap the child with the ErrorCatcher 1`
] = `
<ErrorCatcher>
  <span>
    hello
  </span>
</ErrorCatcher>
`
